src/GitHub.Exports.Reactive/Collections/TrackingCollection.cs

```diff
@@ -0,0 +1,889 @@
+                if (isIncluded)
+                {
+                    // element wasn't included before
+                    if (idx < 0)
+                    {
+                        if (index == Count)
+                            InternalAddItem(item);
+                        else
+                            InternalInsertItem(item, index);
+                    }
+                    index++;
+                }
+                // element is not included and was before
+                else if (idx >= 0)
+                    InternalRemoveItem(item);
+            }
+        }
+
+        /// <summary>
+        /// Get the index in the live list of an object at position.
+        /// This will scan back to the beginning of the live list looking for
+        /// the closest left neighbour and return the position after that.
+        /// </summary>
+        /// <param name="position">The index of an object in the unfiltered, sorted list that we want to map to the filtered live list</param>
+        /// <param name="list">The unfiltered, sorted list of items</param>
+        /// <returns></returns>
+        int GetLiveListPivot(int position, IList<T> list)
+        {
+            var index = -1;
+            if (position > 0)
+            {
+                for (int i = position - 1; i >= 0; i--)
+                {
+                    index = GetIndexFiltered(list[i]);
+                    if (index >= 0)
+                    {
+                        // found an element to the left of what we want, so now we know the index where to start
+                        // manipulating the list
+                        index++;
+                        break;
+                    }
+                }
+            }
+
+            // there was no element to the left of the one we want, start at the beginning of the live list
+            if (index < 0)
+                index = 0;
+            return index;
+        }
+
+        /// <summary>
+        /// Adds an item to the filtered list
+        /// </summary>
+        void InternalAddItem(T item)
+        {
+            isChanging = true;
+            Add(item);
+        }
+
+        /// <summary>
+        /// Inserts an item into the filtered list
+        /// </summary>
+        void InternalInsertItem(T item, int position)
+        {
+            isChanging = true;
+            Insert(position, item);
+        }
+
+        protected override void InsertItem(int index, T item)
+        {
+            if (!isChanging)
+                throw new InvalidOperationException("Collection cannot be changed manually.");
+            isChanging = false;
+
+            filteredIndexCache.Add(item, index);
+            UpdateIndexCache(index, Count, Items, filteredIndexCache);
+            base.InsertItem(index, item);
+        }
+
+        /// <summary>
+        /// Removes an item from the filtered list
+        /// </summary>
+        int InternalRemoveItem(T item)
```

Comment:
The return value is never used. Could we just make this a `void` method since it's private.